### PR TITLE
TODO's addressing

### PR DESCRIPTION
**Pull request by csgui**

### CODE REVIEW

File: clarity/src/vm/clarity_wasm.rs [hunk 1]
```diff
@@ -1566,9 +1554,9 @@ fn pass_argument_to_wasm(
             buffer.push(Val::I32(written));
             Ok((buffer, offset + written, in_mem_offset + in_mem_written))
         }
-        Value::Principal(_p) => todo!("Value type not yet implemented: {:?}", value),
-        Value::CallableContract(_c) => todo!("Value type not yet implemented: {:?}", value),
-        Value::Tuple(_t) => todo!("Value type not yet implemented: {:?}", value),
+        Value::Principal(_p) => Err(Error::Wasm(WasmError::UnhandledType)),
+        Value::CallableContract(_c) => Err(Error::Wasm(WasmError::UnhandledType)),
+        Value::Tuple(_t) => Err(Error::Wasm(WasmError::UnhandledType)),
```
Comment (obycode) [inline clarity/src/vm/clarity_wasm.rs:1559]: Same for these, but don't worry about opening an issue for these, @Acaccia is already working on them.

File: clarity/src/vm/errors.rs [hunk 2]
```diff
@@ -144,6 +144,7 @@ pub enum WasmError {
     ExpectedReturnValue,
     InvalidIndicator(i32),
     Runtime(wasmtime::Error),
+    UnhandledType,
```
Comment (obycode) [inline clarity/src/vm/errors.rs:147]: This error should instead be `InvalidType`. We don't want an error for a type that is not handled yet, just an error for a type that should not be allowed where it was found.

Comment (obycode) [reply clarity/src/vm/errors.rs:147]: Or you could use the existing `InvalidNoTypeInValue` and add `InvalidListUnionTypeInValue`.

File: clarity/src/vm/errors.rs [hunk 3]
```diff
@@ -182,6 +183,7 @@ impl fmt::Display for WasmError {
                 write!(f, "Invalid response/optional indicator: {indicator}")
             }
             WasmError::Runtime(e) => write!(f, "Runtime error: {e}"),
+            WasmError::UnhandledType => write!(f, "Type not handled"),
```
Comment (obycode) [inline clarity/src/vm/errors.rs:186]: Update this accordingly.

**Review by obycode — CHANGES_REQUESTED** [review]

File: clarity/src/vm/clarity_wasm.rs [hunk 4]
```diff
@@ -1530,7 +1518,7 @@ fn pass_argument_to_wasm(
             Ok((buffer, offset, adjusted_in_mem_offset))
         }
         Value::Sequence(SequenceData::String(CharType::UTF8(_s))) => {
-            todo!("Value type not yet implemented: {:?}", value)
+            Err(Error::Wasm(WasmError::UnhandledType))
```
[the remark this answers is not in the document]
Comment (csgui) [reply clarity/src/vm/clarity_wasm.rs:1521]: Ok. Reverted.

File: clarity/src/vm/clarity_wasm.rs (hunk 1 above)
Comment (csgui) [reply clarity/src/vm/clarity_wasm.rs:1559]: Ok. Reverted.

File: clarity/src/vm/errors.rs (hunk 3 above)
Comment (csgui) [reply clarity/src/vm/errors.rs:186]: Done

**Comment by csgui** [comment]
@obycode Made changes to address the errors. Now I understand your intent. Thanks!

**Review by obycode — APPROVED** [review]
👍

**Pull request merged**